02-cpu-configure/cpu-configure.sh

```diff
@@ -1,109 +1,135 @@
+mv /home/lnls-bpm /mnt/@home
+mkdir /mnt/@snapshots/rootfs
+mkdir /mnt/@snapshots/home
+umount /mnt
+mkdir /snapshots
+mount $ROOTFS_PART /home -o subvol=/@home
+mount $ROOTFS_PART /snapshots -o subvol=/@snapshots
+
+# Generate a new fstab to account for the mount points configured
+# previously.
+genfstab -U / > /etc/fstab
+
+# Podman-compose v1.0.3 (in bookworm) mistakenly creates a network bridge
+# and forwards it to podman even with `network_mode` set to `host`, which
+# is used by IOC containers. Providing both options is invalid and causes
+# podman to abort the creation of the container. Using the podman-compose
+# v1.0.6 (in bookworm-backports) fixes this [1].
 #
-## Install packages
-#sudo apt-get update && \
-#sudo apt-get install -y \
-#    openssh-client \
-#    sshpass
-
-
-exec_cmd "TRACE" echo "Generating BPM RFFE IP mapping file..."
-
-# Generate HALCS config from template
-BPM_MAX_NUM_BOARDS=12
-BPM_MAX_NUM_HALCS=2
-cp ${BPM_HALCS_CFG_TEMPLATE_IN_FILE} ${BPM_HALCS_CFG_TEMPLATE_OUT_FILE}
-for board in `seq 1 ${BPM_MAX_NUM_BOARDS}`; do
-    for halcs in `seq 0 $((${BPM_MAX_NUM_HALCS}-1))`; do
-        bpm_rffe_ip_idx="${board}${halcs}"
-        bpm_rffe_ip="BPM_RFFE_IP_MAPPING[${bpm_rffe_ip_idx}]"
-        bpm_rffe_proto="BPM_RFFE_PROTO_MAPPING[${bpm_rffe_ip_idx}]"
-        sed -i \
-            -e "s#<RFFE_BOARD${board}_HALCS${halcs}_IP>#${!bpm_rffe_ip}#" \
-            -e "s#<RFFE_BOARD${board}_HALCS${halcs}_PROTO>#${!bpm_rffe_proto}#" \
-            ${BPM_HALCS_CFG_TEMPLATE_OUT_FILE}
-    done
-done
-
-
-exec_cmd "TRACE" echo "Modifying EPICS PV prefixes..."
-
-# Login via SSH and setup configuration files
-# Set the --static, --transient and --pretty
-# hostnames by the HOSTNAME variable contents
-SSHPASS="${SSHPASS_USR}" sshpass -e \
-    ssh -o StrictHostKeyChecking=no \
-    root@${IP} \
-    bash -c "\
-        echo \"\" > /etc/hostname && \
-        sysctl kernel.hostname=\"${HOSTNAME}\" > /dev/null && \
-        sysctl -w kernel.hostname=\"${HOSTNAME}\" > /dev/null && \
-        hostnamectl set-hostname \"${HOSTNAME}\" > /dev/null && \
-        sed -i -e \"\
-            { \
-                s|EPICS_PV_CRATE_PREFIX=.*\$|EPICS_PV_CRATE_PREFIX=${CRATE}|; \
-            }\" ${BPM_EPICS_CFG_FILE} && \
-        sed -i -e \"\
-            { \
-                s|EPICS_PV_CRATE_PREFIX=.*\$|EPICS_PV_CRATE_PREFIX=${CRATE}|; \
-            }\" ${TIM_RX_EPICS_CFG_FILE} \
-    "
-
-exec_cmd "TRACE" echo "Copying generated files to CPU..."
-
-SSHPASS="${SSHPASS_USR}" sshpass -e \
-    scp \
-    ${BPM_HALCS_CFG_TEMPLATE_OUT_FILE} \
-    root@${IP}:${BPM_HALCS_CFG_FILE}
-
-exec_cmd "INFO " echo "CPU Configured Successfully!"
+# [1]: https://github.com/containers/podman-compose/commit/265e0ca32a7a98c65fbd63f75108ded97bd14fae
+
+printf '
+deb http://deb.debian.org/debian/ bookworm-backports main non-free-firmware contrib non-free
+' >> /etc/apt/sources.list
+apt-get update
+apt-get install podman podman-compose -t bookworm-backports -y
```

Comment:
Missing the `/etc/containers/storage.conf` creation after installing podman. Here goes the description and code for doing so:

```sh
# Configure Podman to use OveralyFS instead of VFS to optimize disk usage, and
# increase performance. It should automatically use overlay filesystem when
# `fuse-overlayfs` is installed [2], which is the case here as `podman` package
# recommends it. However, this automatic discovery did not work during tests,
# falling back to VFS.
#
# [2]: https://docs.podman.io/en/latest/markdown/podman-system-reset.1.html#switching-rootless-user-from-vfs-driver-to-overlay-with-fuse-overlayfs

printf '[storage]

driver = "overlay"
' > /etc/containers/storage.conf
```

@ericonr, do you want to debug this one as well? I gave up because my next steps would involve reading complex Go code; language that I'm not familiar with.

Reply:
given everything, we might move towards not using podman at all. we can take the final decision on tomorrow's meeting

Reply:
Podman removed.